sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/xml/XmlOptions.scala

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql.execution.datasources.xml
+
+import java.nio.charset.StandardCharsets
+
+import org.apache.spark.sql.catalyst.util.{ParseMode, PermissiveMode}
+
+/**
+ * Options for the XML data source.
+ */
+private[xml] class XmlOptions(
+    @transient private val parameters: Map[String, String])
+  extends Serializable {
+
+  def this() = this(Map.empty)
+
+  val charset = parameters.getOrElse("charset", XmlOptions.DEFAULT_CHARSET)
```

Comment:
should document it in https://spark.apache.org/docs/latest/sql-data-sources.html

Reply:
Created a sub-task for documentation: https://issues.apache.org/jira/browse/SPARK-44752